jdk17-ubi9/Dockerfile

```diff
@@ -23,6 +23,9 @@ RUN set -o errexit -o nounset \
     && microdnf install -y \
         # common utilities
         make \
+        curl-minimal \
```

Comment:
💅 Why is it minimal here?

Reply:
`curl-minimal` is used in UBI and Corretto images as they have smaller variants and I see no need to migrate them to larger curl installs without a good justification.

See also:
https://github.com/amazonlinux/amazon-linux-2023/issues/150#issuecomment-1207233037

Note: UBI 9 does not have the `--allowerasing` flag yet and is also sort of on it's way out.

Thus in the interest of consistency and minimal changes I keep UBI and Corretto on `-minimal` and have the other lists be more similar otherwise.